README.md

```diff
@@ -428,9 +501,65 @@ hook.remove(name, hookMethod)
 Example
 
 ```js
-hook.remove('save', validateRecord)
+collection.remove('save', validateRecord)
 ```
 
+## TypeScript
+
+This library contains type definitions for TypeScript. When you use TypeScript we highly recommend using the `Hook.Singular` constructor for your hooks as this allows you to pass along type information for the options object. For example:
+
+```ts
+
+import * as Hook from 'before-after-hook'
+
+interface Foo {
+  bar: string
+  num: number;
+}
+
+const hook = new Hook.Singular<Foo>();
+
+hook.before(function (foo) {
+
+  // typescript will complain about the following mutation attempts
+  foo.hello = 'world'
+  foo.bar = 123
+
+  // yet this is valid
+  foo.bar = 'other-string'
+  foo.num = 123
+})
+
+const foo = hook({bar: 'random-string'}, function(foo) {
+  // handle `foo`
+  foo.bar = 'another-string'
+})
+
+// foo outputs
+{
+  bar: 'another-string',
+  num: 123
+}
+```
+
+An alternative import:
+```ts
+import {Singular} from 'before-after-hook'
+
+const hook = new Singular<{foo: string}>();
+```
+
+
+[gg](#upgrading-to-1.4)
```

Comment:
what does the `gg` mean?

Reply:
if you want to add an invisible anchor, you can add `<a name="upgrading-to-1.4"></a>`

Reply:
It was a debug statement to see in it would work in the deprecation warning url, I removed it now 😉